### ISSUE TITLE
Support multiple levels of scope in the list of variables

### ISSUE THREAD

**Issue by alan-knight** [dart-lang/webdev#537]
Nested functions, class fields, library variables. Doing this really properly is significant work, but can largely be ported from DDT.

**Comment by grouma** [comment]
Closing as I believe this work has been completed. Please re-open if I'm mistaken.

**Comment by alan-knight** [comment]
I don't think it has. Devtools only has pretty limited display of these, but VSCode or Intellij probably could do something useful. But let's open more specific bugs as the need arises. I don't think there's a risk of forgetting this.

**Comment by alan-knight** [comment]
To clarify, by multiple levels of scope, I mean things like

```
foo() {
   var a = 1;
   var b = () {
       var c = 2;
       print(a);
    }
   return b;
}
```
and in the debugger if we're inside an invocation of `b` we'd expect to see not just `c` but also `a` and possibly `b`.

And right now Devtools doesn't show class fields or top-level library variables at all, but others might.
